dotnet-roslyn/analyzers/Audacia.CodeAnalysis.Analyzers/tests/Audacia.CodeAnalysis.Analyzers.Test/Rules/MagicNumberAnalyzerTests.cs

```diff
@@ -312,22 +313,183 @@ private int Calculate(int? arg)
+                    }
+            }
+        }
+    }";
+            var expected = BuildExpectedResult(10, 31, SyntaxKind.WhileStatement.ToString());
+            VerifyDiagnostic(test, expected);
+        }
+
+        [TestMethod]
+        public void Diagnostic_Multi_Variable_While_Iteriable_Magic_Number()
+        {
+
+            var test = @"
+    namespace ConsoleApplication1
+    {
+        class TypeName
+        {
+            private int Calculate(int? arg)
+            {
+                var count = 0;
+
+                while(count < 11 || count == 11)
+                    {
+                        count++;
+                    }
+            }
+        }
+    }";
+            var expected1 = BuildExpectedResult(10, 31, SyntaxKind.WhileStatement.ToString());
+            var expected2 = BuildExpectedResult(10, 46, SyntaxKind.WhileStatement.ToString());
+            VerifyDiagnostic(test, new[] { expected1, expected2 });
+        }
+
+        [TestMethod]
+        public void Diagnostic_If_Magic_Number()
+        {
+
+            var test = @"
+    namespace ConsoleApplication1
+    {
+        class TypeName
+        {
+            private int Calculate(int? arg)
+            {
+                var check = 11;
+
+                if(check == 11)
+                    {
+                    }
+            }
+        }
+    }";
+            var expected = BuildExpectedResult(10, 29, SyntaxKind.IfStatement.ToString());
+            VerifyDiagnostic(test, expected);
+        }
+
+        [TestMethod]
+        public void Diagnostic_Multi_Variable_If_Magic_Number()
+        {
+
+            var test = @"
+    namespace ConsoleApplication1
+    {
+        class TypeName
+        {
+            private int Calculate(int? arg)
+            {
+                var check = 11;
+
+                if(check == 11 && check != 42)
+                    {
+                    }
+            }
+        }
+    }";
+      
+            var expected1 = BuildExpectedResult(10, 29, SyntaxKind.IfStatement.ToString());
+            var expected2 = BuildExpectedResult(10, 44, SyntaxKind.IfStatement.ToString());
+            VerifyDiagnostic(test, new[] { expected1, expected2 });
         }
 
+        [TestMethod]
+        public void Diagnostic_Case_Switch_Label_Magic_Number()
+        {
+
+            var test = @"
+    namespace ConsoleApplication1
+    {
+        class TypeName
+        {
+            private int Calculate(int? arg)
+            {
+                var check = 11;
+
+                 switch (check):
+                {
+                case 11:
+                    return;
+                default:
+                    return;
+                } 
+            }
+        }
+    }";
+
+            var expected = BuildExpectedResult(12, 22, SyntaxKind.CaseSwitchLabel.ToString());
+
+            VerifyDiagnostic(test, expected);
+        }
+
+        [TestMethod]
+        public void Diagnostic_Switch_Statement_Magic_Number()
```

Comment:
What do you think to adding any tests for that there is no diagnostics for when no magic numbers are included in loops, switches etc.?

Reply:
Ah, my thinking was these cases were implicitly tested by doing the "check" variables as part of these tests as it would fail for those variable usages in the same loop, switch etc if there was a problem. However I don't disagree that having explicit tests for these would provide clarity in the cases where there was an issue.

Reply:
Added explicit tests and it already brought up an error so they are already paying off.

Reply:
Nice!